packages/gatsby-cli/src/init-starter.ts

```diff
@@ -341,8 +342,28 @@ export async function initStarter(
   trackCli(`NEW_PROJECT`, {
     starterName: hostedInfo ? hostedInfo.shortcut() : `local:starter`,
   })
-  if (hostedInfo) await clone(hostedInfo, rootPath)
-  else await copy(starterPath, rootPath)
+  if (hostedInfo) {
+    await clone(hostedInfo, rootPath)
+  } else {
+    await copy(starterPath, rootPath)
+  }
+
+  const sitePath = sysPath.resolve(rootPath)
+
+  const sitePackageJson = await fs
+    .readJSON(sysPath.join(sitePath, `package.json`))
+    .catch(() => {
+      reporter.verbose(
+        `Could not read "${sysPath.join(sitePath, `package.json`)}"`
+      )
+    })
+
+  await createServiceLock(sitePath, `metadata`, {
+    name: sitePackageJson?.name || rootPath,
+    sitePath,
+    lastRun: Date.now(),
+  }).then(unlock => unlock?.())
```

Comment:
Yeah, though it would still need most of the same props. e.g. it won't always want to save the pid or update the lastRun (e.g. if it's being called from Desktop)

Reply:
those values could be nullable/undefined maybe?